.github/workflows/publish.yaml

```diff
@@ -0,0 +1,33 @@
+name: publish
+on:
+  push:
+    branches:
+      - "master"
+      - "main"
+    tags:
+      - "v*"
+jobs:
+  image:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Login to image registry
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
+      - name: Generate image names
+        id: images
+        run: |
+          echo -n ::set-output name=IMAGES::
+          tag=($echo "quay.io/observatorium/obsctl:$(git rev-parse --abbrev-ref HEAD | tr / -)-$(date +%Y-%m-%d)-$(git rev-parse --short HEAD)")
+          TAGS=($echo $(echo $tag | awk '{ print tolower($1) }'))
+          ( IFS=$','; echo "${TAGS[*]}" )
+      - name: Build and publish image(s) to registry
+        uses: docker/build-push-action@v2
```

Comment:
I wonder if there is a need for a multi-arch building step here. Especially as obsctl is meant to run on people's machines.

Reply:
People's machines might be an Apple Silicon Macbook (arm64)... it's getting more and more common. 😄

Reply:
+1 multiarch builds pls. Most of the team don't use M1 macs... :{